src/components/RegistrationList.tsx

```diff
@@ -66,6 +88,28 @@ export const RegistrationListItemContent = ({
   const publicationDate = displayDate(entityDescription?.publicationDate);
   const heading = [typeString, publicationDate].filter(Boolean).join(' — ');
 
+  const isPromotedPublication = promotedPublications && promotedPublications.includes(identifier);
+
+  const mutatePromotedPublications = useMutation({
+    mutationFn: () => updatePromotedPublications(promotedPublications ? promotedPublications : []),
+    onSuccess: () =>
+      dispatch(setNotification({ message: t('feedback.success.add_promoted_publication'), variant: 'success' })),
+    onError: () =>
+      dispatch(setNotification({ message: t('feedback.error.add_promoted_publication'), variant: 'error' })),
+  });
+
+  const handleChange = () => {
+    if (!isPromotedPublication) {
+      promotedPublications?.push(identifier);
+    } else {
+      const index = promotedPublications.indexOf(identifier, 0);
+      if (index > -1) {
+        promotedPublications.splice(index, 1);
+      }
+    }
+    mutatePromotedPublications.mutate();
+  };
```

Comment:
Må se litt nærmere på denne litt senere (rekker ikke akkurat nå). Noen ting jeg ikke forstår på stående fot:
- `updatePromotedPublications()` sender tomt array om `promotedPublications`er false. I den casen vil vi fjerne alle favorittmarkeringer. Når/hvorfor kommer man dit, og er det det vi vil?
- som vi har snakket litt om tidligere kan det være skummelt å bruke mutable verdier med `push` og `splice`, da jeg ser for meg at det lett kan komme ut av sync med realiteten. Sånn jeg leser det vil endringene presenteres før kallet gjøres, så om kallet feiler har vi et problem. må se litt nærmere på det for å kunne komme med noen gode forslag da..

Reply:
Prøvde med `mutationFn: () => new Promise<void>((resolve) => resolve())`, og da virker det for brukeren som at alt virker ja, så vi har et problem med consistency her. Usikker på hvordan vi best kan løse det, så må tygge litt på det.. 🤔 
Tror vi må håndtere responsen i `onSuccess`, for å sikre at vi alltid viser den persisterte verdien til brukeren. Da trenger vi nok ikke `handleChange` her lenger. Så får vi se om vi får løst det på en god måte uten mange flere props da 😬

Reply:
tenker også vi burde kalle `mutatePromotedPublications.mutate(newPromotedPublications)` hvor `newPromotedPublications` er arrayet vi ønsker å persistere

Reply:
parprogget oss til en løsning for alt dette :)